eq-author-api/utils/datastore.js

```diff
@@ -1,302 +1,21 @@
-const jsondiffpatch = require("jsondiffpatch");
-const { omit, first, isNil } = require("lodash");
-const logger = require("pino")();
-const uuid = require("uuid");
-const {
-  QuestionnaireModel,
-  QuestionnaireVersionsModel,
-  dynamoose,
-  justListFields,
-  UserModel,
-  CommentsModel,
-} = require("../db/models/DynamoDB");
+const datastoreDynamoDb = require("./datastore-dynamodb");
 
-const { questionnaireCreationEvent } = require("./questionnaireEvents");
+const DYNAMODB = "dynamodb";
 
-const omitTimestamps = questionnaire =>
-  omit({ ...questionnaire }, ["updatedAt", "createdAt"]);
+if (!process.env.DATABASE) {
+  throw new Error("Unset env var 'DATABASE'");
+}
 
-const diffPatcher = jsondiffpatch.create({
-  objectHash: obj => obj.id,
-});
+const databaseName = process.env.DATABASE;
 
-const saveModel = (model, options = {}) =>
-  new Promise((resolve, reject) => {
-    model.save(options, err => {
-      if (err) {
-        reject(err);
-      }
-      resolve(model);
-    });
-  });
+let datastore;
 
-const createUser = user => {
-  const { id, email, name, externalId, picture } = user;
-  return saveModel(
-    new UserModel({
-      id: id || uuid.v4(),
-      email,
-      name,
-      externalId,
-      picture,
-    })
-  );
-};
+if (databaseName === DYNAMODB) {
+  datastore = datastoreDynamoDb;
```

Comment:
I didn't think we should default to a providor, just in case, as otherwise we might encounter some unpredicted behaviour if we don't realise. But, we can talk about this later on to see which is better.